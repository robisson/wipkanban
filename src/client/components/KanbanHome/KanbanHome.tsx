import * as React from "react";
import Grid from '@material-ui/core/Grid';
import BoardList from '../Board/BoardListContainer';
import TeamList from '../Team/TeamList';

const KanbanHome = () => {
  return (
    <Grid container justify="flex-start" wrap="nowrap" spacing={0}>
      <Grid item lg={6}>
        <BoardList/>
      </Grid>
      <Grid item lg={6}>
        <TeamList/>
      </Grid>
    </Grid>
  )
};

export default KanbanHome;
