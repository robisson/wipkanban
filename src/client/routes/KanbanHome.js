// @flow
import * as React from "react";
import Grid from '@material-ui/core/Grid';
import BoardList from './BoardList';
import TeamList from './TeamList';

const KanbanHome = () => {
  return (
    <Grid container justify="flex-start" wrap="nowrap" spacing={0}>
      <Grid item lg={6}>
        <BoardList/>
      </Grid>
      <Grid item lg={6}>
        <TeamList/>
      </Grid>
    </Grid>
  )
};

export default KanbanHome;
