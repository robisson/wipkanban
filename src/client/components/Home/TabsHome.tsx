import React from "react";
import Paper from "@material-ui/core/Paper";
import Tabs from "@material-ui/core/Tabs";
import Tab from "@material-ui/core/Tab";
import PropTypes from "prop-types";
import Typography from "@material-ui/core/Typography";
import KanbanHome from "../KanbanHome";

interface ITabContainerProps {
  children: any;
}

function TabContainer(props: ITabContainerProps) {
  return (
    <Typography component="div" style={{ padding: 8 * 3 }}>
      {props.children}
    </Typography>
  );
}

TabContainer.propTypes = {
  children: PropTypes.node.isRequired
};

interface ITabsHomeState {
  value: number;
}

class TabsHome extends React.Component<{}, ITabsHomeState> {
  state = {
    value: 0
  };

  handleChange = (event: any, value: number) => {
    this.setState({ value });
  };

  render() {
    const { value } = this.state;
    return (
      <div style={{ width: "100%" }}>
        <Paper>
          <Tabs
            value={value}
            onChange={this.handleChange}
            indicatorColor="secondary"
            textColor="primary"
          >
            <Tab label="Kanban boards" />
            <Tab label="OKRs" />
          </Tabs>
        </Paper>
        <div>
          {value === 0 && (
            <TabContainer>
              <KanbanHome />
            </TabContainer>
          )}
          {value === 1 && <TabContainer>ITEM 2</TabContainer>}
        </div>
      </div>
    );
  }
}

export default TabsHome;
