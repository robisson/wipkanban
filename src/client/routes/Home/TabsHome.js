import React from 'react';
import Paper from '@material-ui/core/Paper';
import Tabs from '@material-ui/core/Tabs';
import Tab from '@material-ui/core/Tab';
import PropTypes from 'prop-types';
import Typography from '@material-ui/core/Typography';
import KanbanHome from '../KanbanHome';

function TabContainer(props) {
  return (
    <Typography component="div" style={{ padding: 8 * 3 }}>
      {props.children}
    </Typography>
  );
}

TabContainer.propTypes = {
  children: PropTypes.node.isRequired,
};


class TabsHome extends React.Component {
  state = {
    value: 0
  };

  handleChange = (event, value) => {
    this.setState({value});
  };

  render() {
    const {value} = this.state;
    return (
      <div style={{width:'100%'}}>
        <Paper>
          <Tabs
            value={value}
            onChange={this.handleChange}
            indicatorColor="secondary"
            textColor="primary">
            <Tab label="Kanban boards"/>
            <Tab label="OKRs"/>
          </Tabs>
        </Paper>
        <div>
          {value === 0 && <TabContainer><KanbanHome /></TabContainer>}
          {value === 1 && <TabContainer>ITEM 2</TabContainer>}
        </div>
      </div>
    );
  }
}

export default TabsHome;
