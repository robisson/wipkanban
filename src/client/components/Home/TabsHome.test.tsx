import React from "react";
import { createShallow } from "@material-ui/core/test-utils";
import TabsHome from "./TabsHome";
import Paper from "@material-ui/core/Paper";
import Tabs from "@material-ui/core/Tabs";
import Tab from "@material-ui/core/Tab";
import KanbanHome from "../KanbanHome";

describe("TabsHome component", () => {
  let shallow: any;

  beforeEach(() => {
    shallow = createShallow();
  });

  it("should render 1 div and 1 Paper Component", () => {
    let wrapper = shallow(<TabsHome />);
    expect(wrapper.childAt(0).type()).toEqual(Paper);
    expect(wrapper.childAt(1).type()).toEqual("div");
    expect(wrapper).toMatchSnapshot();
  });

  it("should render 1 Tabs and 2 tabs Component", () => {
    let wrapper = shallow(<TabsHome />);
    expect(wrapper.find(Tabs).type()).toEqual(Tabs);
    expect(wrapper.find(Tab).length).toEqual(2);
    expect(wrapper).toMatchSnapshot();
  });

  it("should render KanbanHome component", () => {
    let wrapper = shallow(<TabsHome />);
    expect(wrapper.find(KanbanHome).type()).toEqual(KanbanHome);
    expect(wrapper).toMatchSnapshot();
  });
});
