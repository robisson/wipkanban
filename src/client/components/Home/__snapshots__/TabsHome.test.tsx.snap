// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabsHome component should render 1 Tabs and 2 tabs Component 1`] = `
<div
  style={
    Object {
      "width": "100%",
    }
  }
>
  <WithStyles(ForwardRef(Paper))>
    <WithStyles(ForwardRef(Tabs))
      indicatorColor="secondary"
      onChange={[Function]}
      textColor="primary"
      value={0}
    >
      <WithStyles(ForwardRef(Tab))
        label="Kanban boards"
      />
      <WithStyles(ForwardRef(Tab))
        label="OKRs"
      />
    </WithStyles(ForwardRef(Tabs))>
  </WithStyles(ForwardRef(Paper))>
  <div>
    <TabContainer>
      <KanbanHome />
    </TabContainer>
  </div>
</div>
`;

exports[`TabsHome component should render 1 div and 1 Paper Component 1`] = `
<div
  style={
    Object {
      "width": "100%",
    }
  }
>
  <WithStyles(ForwardRef(Paper))>
    <WithStyles(ForwardRef(Tabs))
      indicatorColor="secondary"
      onChange={[Function]}
      textColor="primary"
      value={0}
    >
      <WithStyles(ForwardRef(Tab))
        label="Kanban boards"
      />
      <WithStyles(ForwardRef(Tab))
        label="OKRs"
      />
    </WithStyles(ForwardRef(Tabs))>
  </WithStyles(ForwardRef(Paper))>
  <div>
    <TabContainer>
      <KanbanHome />
    </TabContainer>
  </div>
</div>
`;

exports[`TabsHome component should render KanbanHome component 1`] = `
<div
  style={
    Object {
      "width": "100%",
    }
  }
>
  <WithStyles(ForwardRef(Paper))>
    <WithStyles(ForwardRef(Tabs))
      indicatorColor="secondary"
      onChange={[Function]}
      textColor="primary"
      value={0}
    >
      <WithStyles(ForwardRef(Tab))
        label="Kanban boards"
      />
      <WithStyles(ForwardRef(Tab))
        label="OKRs"
      />
    </WithStyles(ForwardRef(Tabs))>
  </WithStyles(ForwardRef(Paper))>
  <div>
    <TabContainer>
      <KanbanHome />
    </TabContainer>
  </div>
</div>
`;
